### DISCUSSION TITLE
Kafka interceptors

### DISCUSSION THREAD

**Discussion by alesj** [open-telemetry/opentelemetry-java-instrumentation#4061] [Q&A]
Does something like this already exist somewhere?
* https://github.com/opentracing-contrib/java-kafka-client/tree/master/opentracing-kafka-client/src/main/java/io/opentracing/contrib/kafka

If not, what would be the best place in OpenTelemetry repos?

Perhaps here:
* https://github.com/open-telemetry/opentelemetry-java-instrumentation/tree/main/instrumentation/kafka-clients/kafka-clients-0.11

but creating new directory -- "intreceptors" or "utils" ?

**Comment by mateuszrzeszutek** [comment]
Hey @alesj,
Creating a new library instrumentation for that would probably be the best choice - i.e. `instrumentation/kafka-clients/kafka-clients-0.11/library`.
